src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,78 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import mate.academy.springboot.datajpa.service.mapper.RequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.ResponseDtoMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final RequestDtoMapper<ProductRequestDto, Product> productRequestDtoMapper;
+    private final ResponseDtoMapper<ProductResponseDto, Product> productResponseDtoMapper;
```

Comment:
Right now, you will have a situation when 2 of these fields will in fact be the same object.
if you want to have it this way, we should not split mappers into 2, because you have only 1 class which implements both mappers so lets add 1 more interface which will extend response and request mappers, and after that our category/product mappers will implement this 3rd interface. this way we will have it as follows:
```suggestion
    private final RequestResponseMapper<ProductResponseDto, Product> productMapper;
```